ui/tests/e2e/fixtures/environmentsPage.js

```diff
@@ -0,0 +1,204 @@
+      // Find the connection status cell in the current row
+      const connectionStatus = await rows
+        .nth(i)
+        .locator('td[data-colindex="10"] .MuiChip-label')
+        .textContent();
+
+      // If the connection status is 'connected', extract the connection name
+      if (connectionStatus.trim().toLowerCase() === 'connected') {
+        const connectionName = await rows
+          .nth(i)
+          .locator('td[data-colindex="3"] .MuiChip-label')
+          .textContent();
+        connectedNames.push({ index: i, name: connectionName.trim() });
+      }
+    }
+
+    // iterate over the connected rows and select the environment for each one
+    for (const { index } of connectedNames) {
+      const dropdownIcon = rows
+        .nth(index)
+        .locator('td[data-colindex="4"] .css-1xc3v61-indicatorContainer')
+        .first();
+
+      // `data-testid` to the SVG icon in the current row
+      await dropdownIcon.evaluate((node, index) => {
+        node.setAttribute('data-testid', `dropdown-svg-${index}`);
+      }, index);
+
+      // Click on the dropdown icon (now identified by `data-testid`)
+      await dropdownIcon.click();
+
+      // Type the environment name into the dropdown input to filter options
+      await this.page.fill(`#react-select-${index}-input`, environmentName);
+
+      // Hit the "Enter" key after selecting the environment option
+      await this.page.press(`#react-select-${index}-input`, 'Enter');
+    }
+  }
+
+  async editEnvironmentCard() {
+    await expect(this.page).toHaveURL(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+    await this.page.click('.MuiCard-root:has-text("Sample-playwright-test") >> :not(:has-text("Assigned Connections"))');
+
+    await this.page.evaluate(() => {
+      const editIcon = document.querySelector(
+        'svg path[d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"]',
+      );
+      if (editIcon) {
+        editIcon.closest('svg').setAttribute('data-testid', 'edit-icon');
+      }
+    });
+    await this.page.getByTestId('edit-icon').click();
+    await this.page.waitForSelector('#Environment_name');
+    await this.fillInput(this.page.locator('#Environment_name'), 'Sample-playwright-test-edited');
+    await this.page.getByRole('button', { name: 'Update', exact: true }).click();
+  }
+
+  async deleteEnvironmentCard() {
+    await expect(this.page).toHaveURL(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+    await this.page.click('.MuiCard-root:has-text("Sample-playwright-test") >> :not(:has-text("Assigned Connections"))');
+
+    await this.page.evaluate(() => {
+      const editIcon = document.querySelector(
+        'svg path[d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"]',
+      );
+      if (editIcon) {
+        editIcon.closest('svg').setAttribute('data-testid', 'delete-icon');
```

Comment:
This dependency injection for Icon is bit fragile and can changes very often, it's better to do it directly to component such as input, button etc

Reply:
Yes, sure. I think this is a more stable approach as it reduces the fragility caused by potential changes in the SVG path. I'll refactor the code accordingly.